modules/gapi/src/compiler/gcompiler.cpp

```diff
@@ -348,12 +358,16 @@ void cv::gimpl::GCompiler::validateInputMeta()
         const auto &meta  = std::get<0>(ade::util::value(meta_arg_idx));
         const auto &proto = std::get<1>(ade::util::value(meta_arg_idx));
 
-        if (!meta_matches(meta, proto))
+        std::stringstream ss;
+        if (!meta_matches(meta, proto, ss))
         {
             const auto index  = ade::util::index(meta_arg_idx);
+
+            const std::string& reason_descr = ss.str();
```

Comment:
`str()` returns `const&` only since C++20, so here there's no need in `&` (two decades earlier it even could result in a hanging reference)

Reply:
fixed